webapp/partners_api.py

```diff
@@ -13,92 +14,53 @@ def get(endpoint):
 
 
 def get_cloud():
-    return get("technology__name=Cloud/server&featured=true")
-
-
-def get_server():
-    return get("technology__name=Cloud/server&featured=true")
-
-
-def get_desktop():
-    return get("programme__name=Desktop&featured=true")
+    return get("programme__name=Certified%20Public%20Cloud&featured=true")
 
 
 def get_silicon():
-    return get("programme__name=Desktop&featured=true")
-
+    return get("service_offered__name=Hardware%20manufacturer&featured=true")
 
-def get_openstack():
-    return get("programme__name=Desktop&featured=true")
 
-
-def get_apps():
-    return get("programme__name=Desktop&featured=true")
+def get_server():
+    return get("service_offered__name=Hardware%20manufacturer&featured=true")
 
 
-def get_snapcraft():
+def get_devices():
     return get("programme__name=Internet%20of%20Things&featured=true")
 
 
-def get_resellers():
-    return get("programme__name=Desktop&featured=true")
-
-
-def get_devices():
-    return get("programme__name=Desktop&featured=true")
+def get_system_integrators():
+    return get("programme__name=Channel&featured=true")
 
 
 def get_charms():
     return get("programme__name=Charm%20Partner%20Programme&featured=true")
 
 
-def get_hosting():
-    return get("service__offered=Hosting%20provider&featured=true")
-
-
-def get_system_integrators():
-    return get("service__offered=System%20integrator/consultant&featured=true")
-
-
-def get_training():
-    return get("programme__name=Desktop&featured=true")
-
-
-def get_kubernetes():
-    return get("programme__name=Desktop&featured=true")
-
-
-def get_paas():
-    return get("programme__name=Desktop&featured=true")
-
-
-def get_serverless():
-    return get("programme__name=Desktop&featured=true")
+def get_apps_and_snaps():
+    return get(
+        "service_offered__name=Software/content%20publisher&featured=true"
+    )
 
 
-def first_ten(req):
-    new_req = req[:10]
+def random_ten(req):
+    if len(req) >= 10:
+        new_req = sample(req, 10)
+    else:
+        new_req = sample(req, len(req))
+    print(len(new_req))
     return new_req
 
 
 def get_partner_groups():
     partner_groups = {
```

Comment:
we have to make one api call?

Here if one fails the entire page will fail.

Reply:
@tbille I agree that one api call would be better, however, in this situation if I make one api call and get all the partners it takes a long time (like 4-5 seconds). I can definitely do that , but let's catch up next week and discuss further about this.